libs/auth-api-lib/src/lib/delegations/delegations-incoming-custom.service.ts

```diff
@@ -84,6 +83,33 @@ export class DelegationsIncomingCustomService {
     })
   }
 
+  async findAllValidGeneralMandate(
+    { nationalId }: FindAllValidIncomingOptions,
+    useMaster = false,
+  ): Promise<DelegationDTO[]> {
+    const { delegations, fromNameInfo } =
+      await this.findAllIncomingGeneralMandates(
+        {
+          nationalId,
+          validity: DelegationValidity.INCLUDE_FUTURE,
+        },
+        useMaster,
+      )
+
+    const delegationDTOs = delegations.map((d) =>
+      d.toDTO(AuthDelegationType.GeneralMandate),
+    )
+
+    return delegationDTOs.map((d) => {
```

Comment:
Here we are looping 2x through the same array, we should join these maps to safe the extra iteration.